### PR TITLE
Add Datadog credential type. Closes #88

### LINKED ISSUE #88: Add Datadog credential type
https://docs.datadoghq.com/agent/guide/environment-variables/
```
credential "datadog" "datadog_creds" {
api_key ="b1cf234................."
app_key ="1a2345bc..................."
api_url = "https://api.adoghq.com/"
}
```

`DD_CLIENT_API_KEY`, `DD_CLIENT_APP_KEY`

### PR DESCRIPTION
**Pull request by karanpopat**
```
Running tool: /opt/homebrew/bin/go test -timeout 30s -run ^TestDatadogDefaultCredential$ github.com/turbot/pipe-fittings/modconfig

ok  	github.com/turbot/pipe-fittings/modconfig	0.456s
```

### CODE REVIEW

File: modconfig/flowpipe_credential.go [hunk 1]
```diff
@@ -1880,6 +1880,84 @@ func (c *BitbucketCredential) Validate() hcl.Diagnostics {
...
+			UnqualifiedName: c.UnqualifiedName,
+			ShortName:       c.ShortName,
+			DeclRange:       c.DeclRange,
+			blockType:       c.blockType,
+		},
+		Type:   c.Type,
+		APIUrl: c.APIUrl,
+	}
+
+	if c.APIKey == nil && datadogAPIKeyEnvVar == "" {
```
[the remark this answers is not in the document]
Comment (karanpopat) [reply modconfig/flowpipe_credential.go]: Removed the validation for now

File: modconfig/flowpipe_credential.go [hunk 2]
```diff
@@ -1880,6 +1880,80 @@ func (c *BitbucketCredential) Validate() hcl.Diagnostics {
...
+			UnqualifiedName: c.UnqualifiedName,
+			ShortName:       c.ShortName,
+			DeclRange:       c.DeclRange,
+			blockType:       c.blockType,
+		},
+		Type:   c.Type,
+		APIUrl: c.APIUrl,
+	}
+
+	if c.APIKey == nil {
```
Comment (vhadianto) [inline modconfig/flowpipe_credential.go:1938]: This is a bug ... what if c.APIKey is NOT null?

File: modconfig/flowpipe_credential.go [hunk 3]
```diff
@@ -1880,6 +1880,80 @@ func (c *BitbucketCredential) Validate() hcl.Diagnostics {
...
+		newCreds.APIKey = &datadogAPIKeyEnvVar
+	}
+
+	if c.AppKey == nil {
```
Comment (vhadianto) [inline modconfig/flowpipe_credential.go:1942]: ditto

File: modconfig/flowpipe_credential.go (hunk 2 above)
Comment (karanpopat) [reply modconfig/flowpipe_credential.go:1938]: ah, added the default in that case

File: modconfig/flowpipe_credential.go (hunk 3 above)
Comment (karanpopat) [reply modconfig/flowpipe_credential.go:1944]: added the default in that case

**Review by vhadianto — APPROVED** [review]

**Pull request merged**